### ISSUE TITLE
scanProg.st crashes Windows IOCs built with Visual Studio 2010

### ISSUE THREAD

**Issue by kmpeters** [epics-modules/sscan#8]
scanProg.st uses the "%T" format for strftime, which is not implemented in Visual Studio 2010.  This results in the IOC crashing when a scan is executed.

Here is some info about the problem:
http://ebanshi.cc/questions/4828844/debug-assertion-failure-while-using-strftime

Replacing "%T" with "%H:%M:%S" should fix the problem.

**Comment by kmpeters** [comment]
Here are the formats that are available strftime when using Visual Studio 2010:
https://msdn.microsoft.com/en-us/library/fe06s4ak%28v=vs.100%29.aspx

"%e" will also need to be replaced.

**Comment by kmpeters** [comment]
Replacing "%e" with "%d" in scanProg.st eliminates the strftime error message, but the IOC still crashes, this time with an unhandled exception.  There is a problem with formatTimeStr() in scanProg.st, but now I have a backtrace, which gives me hope I can use VS2010 to debug the problem.

**Comment by kmpeters** [comment]
After changing the strftime formats to the following:
```
#define TodayFMT "%H:%M:%S Today (%A)"				/* for today use:		"22:23:44 Today (Wednesday)" */
#define TomorrowFMT "%H:%M:%S Tomorrow (%A)"			/* for tomorrow use:	"22:23:44 Tomorrow (Thursday)" */
#define YesterdayFMT "%H:%M:%S Yesterday (%A)"		/* for yesterday use:	"22:23:44 Yesterday (Tuesday)" */
#define GeneralDateFMT "%H:%M:%S  (%a) %b %d, %Y"		/* for all others use:	"22:23:44  (Thu) Nov  8, 2011" */
```
My IOC crashes when formatTimeStr() is called with full=0 for the first time.  formatTimeStr() contains the following code:
```
	if (!full) {
		tm1 = localtime(&final);
		hh = tm1->tm_hour;				/* save time of day */
		mm = tm1->tm_min;
		ss = tm1->tm_sec;

		now = time(NULL);
		tm1 = localtime(&now);
		tm1->tm_hour = hh;				/* make tm1 the same time of day as final, but on today */
		tm1->tm_min = mm;
		tm1->tm_sec = ss;
```
The call to `localtime(&final)` succeeds; a tm struct is returned.  The call to `localtime(&now)` returns a null pointer, despite the fact that "now" has what appears to be an acceptable value.

**Comment by kmpeters** [comment]
Changing "now" from a long to time_t fixes the problem.

**Comment by kmpeters** [comment]
The format-specifier problems can be averted in the future by avoiding the use of formats that are only included in the Single Unix Specification, which carry the (SU) designation on the strftime man page.

**Comment by keenanlang** [comment]
fixed in 85f19e1
